src/google_trace_exporter_client.rs

```diff
@@ -35,6 +36,7 @@ impl GcpCloudTraceExporterClient {
         Ok(Self {
             client,
             google_project_id: google_project_id.to_string(),
+            resource: Resource::default(),
```

Comment:
I think we just need to propagate this resource as `new` parameters, since this is internal structure and shouldn't affect applications externally.

Then in `GcpCloudTraceExporterBuilder.trace_config` we already have `Resource`, so it can be propagated to `GcpCloudTraceExporter` and then to `GcpCloudTraceExporterClient`.

Reply:
Done! Only in `GcpCloudTraceExporter::for_default_project_id()` I still use `Resource::default()`, but since `GcpCloudTraceExporterBuilder` uses the `new` constructor anyway I don't think it's a huge problem. Or is the Exporter itself often used directly?

Reply:
I'll have a look at it myself, seems there was inconsistency in those methods. Thanks for the update!